Add an access to Logstash metrics indices.

mashhurs opened the pull request:
### Description
We have recently implemented a feature with Logstash integration to monitor Logstash (under tech preview). The integration has data streams (`node`, `pipeline`, `plugins`) will be collected under the `metrics-logstash.*`. Telemetry collection (in KB) tasks run with `kibana_system` user which has a limited access to the indices. This change provides an access to the user to query against indices and collect metrics.

#### Unit testing
```
➜  elasticsearch git:(add-read-access-to-logstash-metrics) ./gradlew ':x-pack:plugin:core:test' --tests "org.elasticsearch.xpack.core.security.authz.store.ReservedRolesStoreTests.testKibanaSystemRole" 
```

### Contribution criterias
- Have you signed the [contributor license agreement](https://www.elastic.co/contributor-agreement)? - Yes
- Have you followed the [contributor guidelines](https://github.com/elastic/elasticsearch/blob/main/CONTRIBUTING.md)? - Yes
- If submitting code, have you built your formula locally prior to submission with `gradle check`? - Yes
- If submitting code, is your pull request against main? Unless there is a good reason otherwise, we prefer pull requests against main and will backport as needed. - Backporting to 8.14 series would great
- If submitting code, have you checked that your submission is for an [OS and architecture that we support](https://www.elastic.co/support/matrix#show_os)?
- If you are submitting this code for a class then read our [policy](https://github.com/elastic/elasticsearch/blob/main/CONTRIBUTING.md#contributing-as-part-of-a-class) for that.

mashhurs committed:
Add an access to Logstash metrics indices which Kibana task runners require to collect metrics.

mashhurs committed:
Remove cross cluster privilege from Logstash metrics indices.

elasticsearchmachine commented:
Pinging @elastic/es-security (Team:Security)

slobodanadamovic commented:
@mashhurs Could you please update the description of this PR explaining the need for this change, and also add a test in [ReservedRolesStoreTests#testKibanaSystemRole](https://github.com/elastic/elasticsearch/blob/0d404d974bf8db1684a6e46ef9a2c20d1d13d5f0/x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/security/authz/store/ReservedRolesStoreTests.java#L417)? It can be as simple as asserting that it can read, rollover, etc. Here is what I have in mind: 

```
        Arrays.asList(
            "metrics-logstash." + randomAlphaOfLength(randomIntBetween(0, 13))
        ).forEach((indexName) -> {
            final IndexAbstraction indexAbstraction = mockIndexAbstraction(indexName);
            assertThat(kibanaRole.indices().allowedIndicesMatcher("indices:foo").test(indexAbstraction), is(false));
            assertThat(kibanaRole.indices().allowedIndicesMatcher("indices:bar").test(indexAbstraction), is(false));
            assertThat(
                kibanaRole.indices().allowedIndicesMatcher(TransportDeleteIndexAction.TYPE.name()).test(indexAbstraction),
                is(false)
            );
            assertThat(kibanaRole.indices().allowedIndicesMatcher(GetIndexAction.NAME).test(indexAbstraction), is(true));
            assertThat(
                kibanaRole.indices().allowedIndicesMatcher(TransportCreateIndexAction.TYPE.name()).test(indexAbstraction),
                is(false)
            );
            assertThat(kibanaRole.indices().allowedIndicesMatcher(TransportIndexAction.NAME).test(indexAbstraction), is(false));
            assertThat(kibanaRole.indices().allowedIndicesMatcher(TransportDeleteAction.NAME).test(indexAbstraction), is(false));
            assertThat(kibanaRole.indices().allowedIndicesMatcher(TransportSearchAction.TYPE.name()).test(indexAbstraction), is(true));
            assertThat(kibanaRole.indices().allowedIndicesMatcher(TransportMultiSearchAction.TYPE.name()).test(indexAbstraction), is(true));
            assertThat(kibanaRole.indices().allowedIndicesMatcher(TransportGetAction.TYPE.name()).test(indexAbstraction), is(true));
            assertThat(kibanaRole.indices().allowedIndicesMatcher(READ_CROSS_CLUSTER_NAME).test(indexAbstraction), is(false));
            assertThat(
                kibanaRole.indices().allowedIndicesMatcher(TransportUpdateSettingsAction.TYPE.name()).test(indexAbstraction),
                is(true)
            );
            assertThat(kibanaRole.indices().allowedIndicesMatcher(TransportPutMappingAction.TYPE.name()).test(indexAbstraction), is(true));
            assertThat(kibanaRole.indices().allowedIndicesMatcher(RolloverAction.NAME).test(indexAbstraction), is(true));
        });
    }
```

kc13greiner reviewed (approved):
LGTM!

Reason:  Although `metrics-*` is a data index, it is a patter that is an exception since it is a known/documented collision pattern, see [here](https://www.elastic.co/guide/en/elasticsearch/reference/current/index-templates.html)

mashhurs committed:
Unit test case added for kibana system role to utilize metrics-logstash indices.

Co-authored-by: Slobodan Adamović <slobodan.adamovic@elastic.co>

mashhurs commented:
> @mashhurs Could you please update the description of this PR explaining the need for this change, and also add a test in [ReservedRolesStoreTests#testKibanaSystemRole](https://github.com/elastic/elasticsearch/blob/0d404d974bf8db1684a6e46ef9a2c20d1d13d5f0/x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/security/authz/store/ReservedRolesStoreTests.java#L417)? It can be as simple as asserting that it can read, rollover, etc. Here is what I have in mind:
> 

@slobodanadamovic thank you so much! I have updated the PR description and added unit test case (included you as a co-author).

mashhurs committed:
Checkstyle fix.
